tests3/pgtests.py

```diff
@@ -642,24 +642,25 @@ def test_cursor_messages(self):
 
         # using INFO message level because they are always sent to the client regardless of
         # client_min_messages: https://www.postgresql.org/docs/11/runtime-config-client.html
-        self.cursor.execute("""
-            CREATE OR REPLACE PROCEDURE test_cursor_messages()
-            LANGUAGE plpgsql
-            AS $$
-            BEGIN
-                RAISE INFO 'hello world' USING ERRCODE = '01000';
-            END;
-            $$;
-        """)
-        self.cursor.execute("CALL test_cursor_messages();")
-        self.assertTrue(type(self.cursor.messages) is list)
-        self.assertEqual(len(self.cursor.messages), 1)
-        self.assertTrue(type(self.cursor.messages[0]) is tuple)
-        self.assertEqual(len(self.cursor.messages[0]), 2)
-        self.assertTrue(type(self.cursor.messages[0][0]) is str)
-        self.assertTrue(type(self.cursor.messages[0][1]) is str)
-        self.assertEqual('[01000] (-1)', self.cursor.messages[0][0])
-        self.assertTrue(self.cursor.messages[0][1].endswith('hello world'))
+        for message in ('hello world', 'A' * 30000):
```

Comment:
Similar to above, just trying to get the tests right:
```python
        for msg in ('hello world', 'ABCDEFGHIJ' * 800):
            self.cursor.execute("""
                CREATE OR REPLACE PROCEDURE test_cursor_messages()
                LANGUAGE plpgsql
                AS $$
                BEGIN
                    RAISE INFO '{}' USING ERRCODE = '01000';
                END;
                $$;
            """.format(msg))
            self.cursor.execute("CALL test_cursor_messages();")
            messages = self.cursor.messages
            self.assertTrue(type(messages) is list)
            self.assertTrue(len(messages) > 0)
            self.assertTrue(all(type(m) is tuple for m in messages))
            self.assertTrue(all(len(m) == 2 for m in messages))
            self.assertTrue(all(type(m[0]) is str for m in messages))
            self.assertTrue(all(type(m[1]) is str for m in messages))
            self.assertTrue(all(m[0] == '[01000] (-1)' for m in messages))
            self.assertTrue(''.join(m[1] for m in messages).endswith(msg))
```
Note the "800" here (as opposed to "400" for `tests2`)